### ISSUE TITLE
gdal 2.3 failing tests

### ISSUE THREAD

**Issue by rbuffat** [Toblerity/Fiona#573]
I was looking at why the travis builds against gdal 2.3 are failing.

When building gdal on travis timeouts after 10min when no compilation output is displayed with the silent option `make -s`. 

test_bytescollection.py -> test_grenada_bytes_geojson does not raise a FionaValueError anymore but on a first glance seems to detect that it json now. This could be related to https://trac.osgeo.org/gdal/wiki/rfc70_output_format_guess

The other failing tests seem to be related to the following behavior change of gdal: The tests fail with "Missing url parameter": https://travis-ci.org/Toblerity/Fiona/jobs/366895108#L2927
 
For http URLs, fiona currently creates the following path: `/vsicurl/svn.osgeo.org/gdal/trunk/autotest/ogr/data/poly.shp`

With gdal 2.2.2, this works:
`ogrinfo /vsicurl/svn.osgeo.org/gdal/trunk/autotest/ogr/data/poly.shp`

with  GDAL 2.3.0dev-9a563c5-dirty, released 2018/04/13:
```
/opt/gdal23/bin/ogrinfo /vsicurl/svn.osgeo.org/gdal/trunk/autotest/ogr/data/poly.shp
ERROR 5: Missing url parameter
ERROR 5: Missing url parameter
ERROR 5: Missing url parameter
ERROR 5: Missing url parameter
ERROR 5: Missing url parameter
ERROR 5: Missing url parameter                                                                                                                                                                                                                     
ERROR 5: Missing url parameter                                                                                                                                                                                                                     
ERROR 5: Missing url parameter                                                                                                                                                                                                                     
ERROR 5: Missing url parameter                                                                                                                                                                                                                     
ERROR 5: Missing url parameter                                                                                                                                                                                                                     
ERROR 1: Only read-only mode is supported for /vsicurl                                                                                                                                                                                             
FAILURE:                                                                                                                                                                                                                                           
Unable to open datasource `/vsicurl/svn.osgeo.org/gdal/trunk/autotest/ogr/data/poly.shp' with the following drivers.     
```

While `/opt/gdal23/bin/ogrinfo /vsicurl/http://svn.osgeo.org/gdal/trunk/autotest/ogr/data/poly.shp`
works. I did not test if we would modify the vsicurl paths to include the protocol would solve the issue, but I suspect so.

@sgillies  Was the protocol removed from the vsicurl uri on purpose or is this an upstream bug?

**Comment by sgillies** [comment]
@rbuffat I suspect there is a regression in Fiona. I haven't intended that we drop the protocol as in `/vsicurl/svn.osgeo.org/gdal/trunk/autotest/ogr/data/poly.shp`. GDAL 2.3 might just be less forgiving since adding new vsicurl parameterization.

**Comment by snorfalorpagus** [comment]
Closed by #576.
